kliff/transforms/configuration_transforms/graphs/generate_graph.py

```diff
@@ -0,0 +1,218 @@
+        """
+        A Pytorch Geometric compatible graph representation of a configuration. When loaded
+        into a class:`torch_geometric.data.DataLoader` the graphs of type PyGGraph
+        will be automatically collated and batched.
+
+        """
+
+        def __init__(self):
+            super().__init__()
+            self.num_nodes = (
+                None  # Simplify sizes and frees up pos key word, coords is cleaner
+            )
+            self.energy = None
+            self.forces = None
+            self.n_layers = None
+            self.coords = None
+            self.images = None
+            self.species = None
+            self.z = None
+            self.contributions = None
+
+        def __inc__(self, key, value, *args, **kwargs):
+            if "index" in key or "face" in key:
+                return self.num_nodes
+            elif "contributions" in key:
+                return 2
+            elif "images" in key:
+                return torch.max(value) + 1
+            else:
+                return 0
+
+        def __cat_dim__(self, key, value, *args, **kwargs):
+            if "index" in key or "face" in key:
+                return 1
+            else:
+                return 0
+
+else:
+
+    class PyGGraph:
+        """
+        A dummy class to when torch geometric is not available. It has same attributes
+        as PyGGraph class, but does not inherit from torch_geometric.data.Data.
+        """
+
+        warning_logged = False
+
+        def __init__(self, *args, **kwargs):
+            if not PyGGraph.warning_logged:
+                logger.warning(
+                    "Pytorch Geometric not available, using dummy PyGGraph."
+                    "Install torch geometric for better performance."
+                )
+                PyGGraph.warning_logged = True
+            self.num_nodes = None
+            self.energy = None
+            self.forces = None
+            self.n_layers = None
+            self.coords = None
+            self.images = None
+            self.species = None
+            self.z = None
+            self.contributions = None
+
+
+class KIMDriverGraph(ConfigurationTransform):
+    """
+    Generate a graph representation of a configuration. This generator will also save the
+    required parameters for porting the model over to KIM-API using TorchMLModelDriver.
+    The configuration file saved here will generate identical graphs at KIM-API runtime.
+    For porting the graph representation you also need to provide the TorchScript model file name.
+
+    Args:
+        species (list): List of species.
+        cutoff (float): Cutoff distance.
+        n_layers (int): Number of convolution layers.
+        copy_to_config (bool): If True, the fingerprint will be copied to
+            the Configuration object's fingerprint attribute.
+    """
+
+    def __init__(
+        self,
+        species,
+        cutoff,
+        n_layers,
+        copy_to_config=False,
```

Comment:
This is not descriptor and I believe there is nothing to copy. So, remove it in __init__, and set it to `False` in super()?

Update:
Well, we can attach a graph to a config in principle. I'm OK to keep this. But please update the docstring above.

Reply:
Can you clarify which portion of docstring you think needs changing? Updated fingerprint -> graph.

Reply:
I was not talking about docs, but the real code, `copy_to_config=False`. I think the current setting is good. In principle, one can attach a graph to a config.

Reply:
So are there no more changes pending here? Shall I mark it resolved?

Reply:
yes. Marked it.